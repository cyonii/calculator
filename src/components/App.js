import Display from './Display';
import ButtonPanel from './ButtonPanel';
import calculate from '../logic/calculate';

const App = () => (
  <div className="container text-center">
    <h1 className="mt-3 mb-4">Calculator</h1>
    <div className="row justify-content-center">
      <div className="col-sm-8 col-md-4">
        <div className="card calc-card border-0">
          <div className="card-body py-5">
            <Display />
            <ButtonPanel />
          </div>
        </div>
      </div>
    </div>
  </div>
);

export default App;
