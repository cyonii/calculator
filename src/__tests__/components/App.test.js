import { render, screen, within } from '@testing-library/react';
import App from '../../components/App';

describe('<App /> Component', () => {
  it('should render successfully', () => {
    render(<App />);
    const app = screen.getByTestId('app');
    expect(app).toBeInTheDocument();
    expect(within(app).getByTestId('display')).toBeInTheDocument();
    expect(within(app).getByTestId('button-panel')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    render(<App />);
    const app = screen.getByTestId('app');
    expect(app).toMatchSnapshot();
  });
});
