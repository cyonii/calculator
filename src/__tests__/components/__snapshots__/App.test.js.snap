// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Component should match snapshot 1`] = `
<div
  class="container text-center"
  data-testid="app"
>
  <h1
    class="mt-3 mb-4"
  >
    Calculator
  </h1>
  <div
    class="row justify-content-center"
  >
    <div
      class="col-sm-10 col-md-7 col-lg-5 col-xl-4"
    >
      <div
        class="card calc-card border-0"
      >
        <div
          class="card-body py-5"
        >
          <div
            class="px-2 py-3 py-lg-4 rounded-pill mb-3 bg-secondary text-end fs-5 text-white"
            data-testid="display"
          >
            0
          </div>
          <div
            data-testid="button-panel"
          >
            <div
              class="row mb-3"
            >
              <div
                class="col-3"
              >
                <button
                  class="btn  calc-btn w-100 border rounded-pill fs-3 fw-bold py-2"
                  data-testid="btn-AC"
                  type="button"
                >
                  AC
                </button>
              </div>
              <div
                class="col-3"
              >
                <button
                  class="btn  calc-btn w-100 border rounded-pill fs-3 fw-bold py-2"
                  data-testid="btn-+/-"
                  type="button"
                >
                  +/-
                </button>
              </div>
              <div
                class="col-3"
              >
                <button
                  class="btn  calc-btn w-100 border rounded-pill fs-3 fw-bold py-2"
                  data-testid="btn-%"
                  type="button"
                >
                  %
                </button>
              </div>
              <div
                class="col-3"
              >
                <button
                  class="btn btn-primary border-primary calc-btn w-100 border rounded-pill fs-3 fw-bold py-2"
                  data-testid="btn-รท"
                  type="button"
                >
                  รท
                </button>
              </div>
            </div>
            <div
              class="row mb-3"
            >
              <div
                class="col-3"
              >
                <button
                  class="btn  calc-btn w-100 border rounded-pill fs-3 fw-bold py-2"
                  data-testid="btn-7"
                  type="button"
                >
                  7
                </button>
              </div>
              <div
                class="col-3"
              >
                <button
                  class="btn  calc-btn w-100 border rounded-pill fs-3 fw-bold py-2"
                  data-testid="btn-8"
                  type="button"
                >
                  8
                </button>
              </div>
              <div
                class="col-3"
              >
                <button
                  class="btn  calc-btn w-100 border rounded-pill fs-3 fw-bold py-2"
                  data-testid="btn-9"
                  type="button"
                >
                  9
                </button>
              </div>
              <div
                class="col-3"
              >
                <button
                  class="btn btn-primary border-primary calc-btn w-100 border rounded-pill fs-3 fw-bold py-2"
                  data-testid="btn-*"
                  type="button"
                >
                  *
                </button>
              </div>
            </div>
            <div
              class="row mb-3"
            >
              <div
                class="col-3"
              >
                <button
                  class="btn  calc-btn w-100 border rounded-pill fs-3 fw-bold py-2"
                  data-testid="btn-4"
                  type="button"
                >
                  4
                </button>
              </div>
              <div
                class="col-3"
              >
                <button
                  class="btn  calc-btn w-100 border rounded-pill fs-3 fw-bold py-2"
                  data-testid="btn-5"
                  type="button"
                >
                  5
                </button>
              </div>
              <div
                class="col-3"
              >
                <button
                  class="btn  calc-btn w-100 border rounded-pill fs-3 fw-bold py-2"
                  data-testid="btn-6"
                  type="button"
                >
                  6
                </button>
              </div>
              <div
                class="col-3"
              >
                <button
                  class="btn btn-primary border-primary calc-btn w-100 border rounded-pill fs-3 fw-bold py-2"
                  data-testid="btn--"
                  type="button"
                >
                  -
                </button>
              </div>
            </div>
            <div
              class="row mb-3"
            >
              <div
                class="col-3"
              >
                <button
                  class="btn  calc-btn w-100 border rounded-pill fs-3 fw-bold py-2"
                  data-testid="btn-1"
                  type="button"
                >
                  1
                </button>
              </div>
              <div
                class="col-3"
              >
                <button
                  class="btn  calc-btn w-100 border rounded-pill fs-3 fw-bold py-2"
                  data-testid="btn-2"
                  type="button"
                >
                  2
                </button>
              </div>
              <div
                class="col-3"
              >
                <button
                  class="btn  calc-btn w-100 border rounded-pill fs-3 fw-bold py-2"
                  data-testid="btn-3"
                  type="button"
                >
                  3
                </button>
              </div>
              <div
                class="col-3"
              >
                <button
                  class="btn btn-primary border-primary calc-btn w-100 border rounded-pill fs-3 fw-bold py-2"
                  data-testid="btn-+"
                  type="button"
                >
                  +
                </button>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="col-6"
              >
                <button
                  class="btn  calc-btn w-100 border rounded-pill fs-3 fw-bold py-2"
                  data-testid="btn-0"
                  type="button"
                >
                  0
                </button>
              </div>
              <div
                class="col-3"
              >
                <button
                  class="btn  calc-btn w-100 border rounded-pill fs-3 fw-bold py-2"
                  data-testid="btn-."
                  type="button"
                >
                  .
                </button>
              </div>
              <div
                class="col-3"
              >
                <button
                  class="btn btn-primary border-primary calc-btn w-100 border rounded-pill fs-3 fw-bold py-2"
                  data-testid="btn-="
                  type="button"
                >
                  =
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
